Parser/tokenizer.c

```diff
@@ -874,8 +874,12 @@ tok_nextc(struct tok_state *tok)
                 strcpy(newtok, buf);
                 Py_DECREF(u);
             }
-            if (tok->nextprompt != NULL)
-                tok->prompt = tok->nextprompt;
+            if (tok->nextprompt != NULL) {
+                if (newtok != NULL && (*newtok == '#' || *newtok == ' ' || *newtok == '\t'))
```

Comment:
Suggestion: You may be able to simplify this with `strchr` (check that is not slower....maybe is faster if inlined).

Reply:
Thank you for suggestions, I'll apply them.